binder/src/main/java/io/grpc/binder/internal/BinderClientTransport.java

```diff
@@ -316,14 +316,14 @@ void notifyTerminated() {
       readyTimeoutFuture.cancel(false);
       readyTimeoutFuture = null;
     }
-    if (preAuthResultFuture != null) {
-      preAuthResultFuture.cancel(false); // No effect if already complete.
-    }
-    if (authResultFuture != null) {
-      authResultFuture.cancel(false); // No effect if already complete.
-    }
+
     serviceBinding.unbind();
     clientTransportListener.transportTerminated();
+
+    cancelAsync(preAuthResultFuture);
```

Comment:
Did you move these for a reason? Want to leave them where they were, to minimize the diff and to keep all futures together?